src/lib/ngsi/ContextAttribute.cpp

```diff
@@ -357,18 +357,28 @@ std::string ContextAttribute::toJson(bool isLastElement)
 {
   std::string  out;
   bool         isNumber = false;
+  bool         isString = true;
 
   if (type == "number")
   {
     isNumber = true;
   }
 
-  if (((type == "") || (isNumber == true)) && (metadataVector.size() == 0))
+  if(type == "string")
```

Comment:
We are going to have booleans as well. 

I think the current rendering code could be simplified a lot , even we could consider using switch case avoiding too many ifs but that could be part of another PR IMHO and when Ken is here 

NTC